build/testing/applitools/run-msft-docs.js

```diff
@@ -0,0 +1,158 @@
+"use strict";
+
+let spawn = require('child_process').spawnSync;
+
+/**
+ * Get Branch Name using child_process and Git which runs Asynchronously
+ */
+class Config {
+
+    constructor() {
+        this.DomainName = "https://msft-docs.fast-dna.net";
+        this.AppName = "FAST-DNA"; // Should be batch name
+        this.SiteName = "MSFT Doc Site"; // Should be branch name
+    }
+
+    branchName() {
+
+        return spawn('git', ['rev-parse', '--abbrev-ref', 'HEAD']).output[1].toString().trim();
+    
+    }
+}
+exports.Config = Config;
+
+
+/**
+ * Testing script that can be used across services  
+ */
+class Run {
+
+    constructor(eyes, driver, website) {
+
+        console.log("Starting tests on: %s", website);
+
+            driver.get(website);
+
+            // eyes.checkWindow("Before any action");
+            // Note: There are many ways to nagivate/find elements on a page
+            // unfortunately, not all are cross-browser compliant and so
+            // for example, we can't use xpath. As a result, we're using
+            // navigation though not necessarily most performant but effective.
+
+            // After browsers matrix is determined we can fine tune and
+            // optimize to use other selectors to see what works. If we're
+            // Testing on modern browsers this isn't as much a limitation.
+
+            // Recommendation, add in data-test attributes to each component,
+            // page, layout, or container we want to capture and select on that.
+
+            // Iterate each components documentation
+
+            driver.navigate().to(website + "/components/button/");
+            eyes.checkWindow("button");

```

Comment:
What happens when there is a new component added? Does this have to be updated every time? Is there a better way we can think of automating this based on example files which exist?

Reply:
This is the initial implementation and for guidance.  There are a lot of improvements that can be made at the time a comprehensive functional and UI testing plan is created.  To this date we don't have one nor do we have the need.  

If a new component is added and the contributor wants to use this, yes, they should update to add their component.